web/pages/notifications.tsx

```diff
@@ -69,60 +167,326 @@ export default function Notifications() {
+  const [activitySummaryLines, setActivitySummaryLines] = useState<string[]>([])
+  const numSummaryLines = 2
+
+  const summarizeMostRecentActivity = async (notifications: Notification[]) => {
+    let questionUpdates = 0
+    const tempSummaryLines: string[] = []
+    for (let i = 0; i < notifications.length; i++) {
+      if (tempSummaryLines.length + questionUpdates >= numSummaryLines) break
+      const notification = notifications[i]
+      if (!notification) continue
+      const { sourceType, sourceContractId, sourceId } = notification
+      if (!sourceId || !sourceContractId) continue
+      if (sourceType === 'comment' || sourceType === 'answer') {
+        const summary = await GetNotificationSummaryText(
+          sourceId,
+          sourceContractId,
+          sourceType
+        )
+        tempSummaryLines.push(
+          sourceType === 'answer' ? `New answer: ${summary}` : summary
+        )
+      } else if (sourceType === 'contract') {
+        questionUpdates += 1
+      }
+    }
+    if (questionUpdates > 0)
+      tempSummaryLines.push(
+        `${questionUpdates} new question update${
+          questionUpdates > 1 ? 's' : ''
+        }`
+      )
+    setActivitySummaryLines(tempSummaryLines)
+  }
+
+  useEffect(() => {
+    if (!contract) return
+    setNotificationsAsSeen(notifications)
+    summarizeMostRecentActivity(notifications)
+  }, [contract, notifications])
+
+  return (
+    <div className={clsx('bg-white px-2 pt-6 text-sm sm:px-4', className)}>
+      <Row className={'items-center text-gray-500 sm:justify-start'}>
+        <div className="flex h-8 w-8 items-center justify-center rounded-full bg-gray-200">
+          <UsersIcon className="h-5 w-5 text-gray-500" aria-hidden="true" />
+        </div>
+        <div className={'flex-1 overflow-hidden pl-2 sm:flex'}>
+          <div
+            className={
+              'flex max-w-sm shrink overflow-hidden text-ellipsis sm:max-w-md'
+            }
+          >
+            <a
+              href={contract && `${contract.creatorUsername}/${contract.slug}`}
+              className={
+                'inline-flex overflow-hidden text-ellipsis pl-1 sm:pl-0'
+              }
+            >
+              {'Activity on '}
+              {contract?.question ?? 'question'}
+            </a>
+          </div>
+          <RelativeTimestamp time={notifications[0].createdTime} />
+        </div>
+      </Row>
+      <a href={contract && `${contract.creatorUsername}/${contract.slug}`}>
+        <div className={'mt-1 md:text-base'}>
+          {' '}
+          <div className={'line-clamp-4 mt-1 gap-1 whitespace-pre-line'}>
+            {activitySummaryLines.map((line, i) => {
+              return (
+                <div key={line} className={'line-clamp-1'}>
+                  <Linkify text={line} />
+                </div>
+              )
+            })}
+            <div>
+              {notifications.length - numSummaryLines > 0
```

Comment:
No way to expand the group of notifications?

Reply:
will do, thanks!